File: src/main/java/mods/orca/mffs/util/BlockUtils.java
package mods.orca.mffs.util;

import net.minecraft.block.Block;
import net.minecraft.init.Blocks;
import net.minecraft.item.ItemBlock;
import net.minecraft.item.ItemStack;

public class BlockUtils {

    public static Block getBlock(ItemStack itemStack) {
        if (itemStack == null || itemStack.isEmpty() || !(itemStack.getItem() instanceof ItemBlock)) return Blocks.AIR;
        return ((ItemBlock) itemStack.getItem()).getBlock();
    }

    public static boolean isValidCamo(Block block) {
        return block.getBlockState().getBaseState().isFullBlock();
    }

    public static boolean isValidCamo(ItemStack itemStack) {
        return isValidCamo(getBlock(itemStack));
    }
}

File: src/main/java/mods/orca/mffs/blocks/utils/BlockUtils.kt
package mods.orca.mffs.blocks.utils

import net.minecraft.block.Block
import net.minecraft.init.Blocks
import net.minecraft.item.ItemBlock
import net.minecraft.item.ItemStack

/**
 * Get the block from a given item stack if it is an [ItemBlock], or [Blocks.AIR].
 */
fun ItemStack?.getBlock(): Block = this
    ?.takeIf { !it.isEmpty }
    ?.let { it.item as? ItemBlock }
    ?.block
    ?: Blocks.AIR

// TODO
//private fun isValidCamo(block: Block): Boolean {
//    return block.blockState.baseState.isFullBlock
//}
//
///**
// * Get whether the given block is valid for camouflage.
// */
//fun isValidCamo(itemStack: ItemStack?): Boolean {
//    return isValidCamo(itemStack.getBlock())
//}
